test/test_auth_adapter.rb

```diff
@@ -0,0 +1,11 @@
+require 'test_helper'
+
+class TestAuthAdapter < Test::Unit::TestCase
+  def test_undefined_auth_adapter
+    flexmock(TCPSocket).should_receive(:new).ordered.with('ldap.example.com', 379).once.and_return(nil)
```

Comment:
It's too bad that `Connection.new` actually tries to open the socket and forces us to stub here. Not suggesting any change, but making an observation.